### ISSUE TITLE
No lighthouse report is being generated

### ISSUE THREAD

**Issue by reinier** [netlify/netlify-plugin-lighthouse#548]
**Describe the bug**
When building the site in repo [reinierladan.nl](https://github.com/reinier/reinierladan.nl), in the _deploy summary_ I get:

`@netlify/plugin-lighthouse ran successfully`
`Summary for path '_site/': undefined`

In the logs I see minimal activity from the Lighthouse plugin. It installs and does this after:

```
4:08:19 PM:   2. @netlify/plugin-lighthouse (onPostBuild event)             
4:08:19 PM: (@netlify/plugin-lighthouse onPostBuild completed in 8ms)
4:08:20 PM: Post processing - header rules
```

I have the same setup running for repo [reinier.fyi](https://github.com/reinier/fyi) and that works pretty well.

There is also no report generated and no Lighthouse UI visible in the Deploy details.

**Comment by reinier** [comment]
I've setup the netlify.toml here:
https://github.com/reinier/reinierladan.nl/blob/main/netlify.toml

And I've also activated the Lighthouse integration from within Netlify to see if that solves it. It does not unfortunately.

**Comment by intrepidws** [comment]
I also have this issue, but only with some of my sites. Some work and some don't - seemingly without reason. I have tried uninstalling and reinstalling the integration - which didn't work.

**Comment by intrepidws** [comment]
@reinier I was able to fix this, for at least one of my sites, by specifying an env variable for `NODE_VERSION` to be higher than 12. In my case, both 14 and 16 worked. I found #500 and that tipped me off that this might work.

**Comment by reinier** [comment]
You are right @intrepidws. I added an environment variable in Netlify with name NODE_VERSION and value 16 to the site that didn't generated a lighthouse report and now it works!

**Comment by rodrigoalcarazdelaosa** [comment]
I'm not deploying my site with Netlify anymore but [here you can see what I meant](https://answers.netlify.com/t/enable-lighthouse-score-visualizations-in-labs/54004/10). The solution was to install the plugin via the `netlify.toml` file.

**Comment by aitchiss** [comment]
Thank you @rodrigoalcarazdelaosa for sharing the additional context! This is really helpful for us to know, even if you are unblocked for now - appreciate you taking the time ❤️ 

I'll go ahead and close this issue, but if anyone has similar trouble in future please don't hesitate to re-open or create a new issue.

**Comment by reinier** [comment]
Hi @aitchiss, maybe you could check my repo to further investigate. I thought the NODE_VERSION var solved the issue, but it only did it once. My current builds lack any Lighthouse info, even though I think everything is setup correctly.

**Comment by danieljamesross** [comment]
I get similar behaviour. I have the plugin installed on several sites and it works on all but one. They are all Nextjs and all built with the same commands, yet one of them producs no lighthouse output.

**Comment by aitchiss** [comment]
@danieljamesross @jodiedoubleday is it possible to link us to an example deploy, or to your repo (if privacy allows)?

**Comment by jodiedoubleday** [comment]
Hi @aitchiss my repo on GitHub is private but if you are part of the Netlify Team then you might be able to see this deployment - https://app.netlify.com/sites/simon-community/deploys/643eb32ba785270008bd29a7

**Comment by danieljamesross** [comment]
> @danieljamesross @jodiedoubleday is it possible to link us to an example deploy, or to your repo (if privacy allows)?

Afraid not, it's completely private

**Comment by aitchiss** [comment]
Thank you all for the further details - we'll investigate and try to get to the bottom of this - having the couple of different examples helps so thanks again!

**Comment by dylankenneally** [comment]
I have the same issue, and my repo is public if you're looking for more data on the source of this issue.

- Repo: https://github.com/dylankenneally/http-statuses
- `NODE_VERSION` set in [netlify.toml](https://github.com/dylankenneally/http-statuses/blob/master/netlify.toml#L9)
- Plugin set in [netlify.toml](https://github.com/dylankenneally/http-statuses/blob/master/netlify.toml#L17) too
- [Logs](https://app.netlify.com/sites/httpstatuses/deploys/6441c2fed3fa99000728e195) don't seem to be showing anything handy

```
1 plugin ran successfully

Select for details.

@netlify/plugin-lighthouse
@netlify/plugin-lighthouse ran successfully

Summary for path 'build/': undefined
```

**Comment by aitchiss** [comment]
Thanks again everyone for volunteering debug information ❤️ 

Please can you try upgrading to `v4.1.1` of the plugin and checking if your issues have been resolved? I verified the fix locally against a clone of both @reinier and @dylankenneally's repos, but please do re-open this issue if you still encounter the same problems after upgrade.

**Comment by danieljamesross** [comment]
Just upgraded and re-deployed:

<img width="1185" alt="Screenshot 2023-04-21 at 15 04 20" src="https://user-images.githubusercontent.com/28922296/233656551-04f441ab-73e5-49ce-9e96-3d5096dd3e24.png">

**Comment by aitchiss** [comment]
At the moment the lighthouse plugin runs on a static version of the site served on a local express server, which causes lots of challenges with sites where the build process doesn't result in simple static `html` files in the build directory. However - we do have some experimental functionality in place that will run the audit instead on your live deploy. There are lots of benefits to this (eventually we expect it to become the default plugin behaviour), but I bring it up because if the issues you're seeing are related to relying on SSR the new functionality might solve it.

If you'd like to give that new functionality a spin and see if it unblocks things - [check out the instructions here](https://github.com/netlify/netlify-plugin-lighthouse/issues/566#issuecomment-1508235754)

**Comment by danieljamesross** [comment]
OK, this works for me now. Thanks very uch for your help!

**Comment by reinier** [comment]
Hi @aitchiss, thanks for digging into this. I see more happening in my logs, but still no report (repo https://github.com/reinier/reinierladan.nl). This is the part of the build log that covers the lighthouse plugin:

6:29:51 PM:   2. @netlify/plugin-lighthouse (onPostBuild event)             
6:29:51 PM: ────────────────────────────────────────────────────────────────
6:29:51 PM: Starting post processing
6:29:51 PM: ​
6:29:51 PM: Generating Lighthouse report. This may take a minute…
6:29:51 PM: Running Lighthouse on _site/
6:29:51 PM: Post processing - HTML
6:29:51 PM: Error: Could not find local browser
6:29:51 PM:     at getBrowserPath (file:///opt/build/repo/node_modules/@netlify/plugin-lighthouse/src/lighthouse.js:10:11)
6:29:51 PM:     at async runAuditWithServer (file:///opt/build/repo/node_modules/@netlify/plugin-lighthouse/src/lib/run-audit-with-server/index.js:19:25)
6:29:51 PM:     at async runEvent (file:///opt/build/repo/node_modules/@netlify/plugin-lighthouse/src/lib/run-event/index.js:73:9)
6:29:51 PM:     at async onPostBuild (file:///opt/build/repo/node_modules/@netlify/plugin-lighthouse/src/index.js:37:9)
6:29:51 PM: Post processing - header rules
6:29:51 PM:     at async Object.run (file:///opt/buildhome/node-deps/node_modules/@netlify/build/lib/plugins/child/run.js:14:5)
6:29:51 PM:     at async handleEvent (file:///opt/buildhome/node-deps/node_modules/@netlify/build/lib/plugins/child/main.js:28:42)
6:29:51 PM:     at async process.<anonymous> (file:///opt/buildhome/node-deps/node_modules/@netlify/build/lib/plugins/ipc.js:91:24)
6:29:51 PM: Post processing - redirect rules
6:29:51 PM: ​
6:29:51 PM: (@netlify/plugin-lighthouse onPostBuild completed in 11ms)
6:29:51 PM: 

My site is fully static.

**Comment by hardyjohnson** [comment]
@aitchiss 4.1.1 seems to have resolved the issue for our Gatsby site. Thank you.

**Comment by dylankenneally** [comment]
@aitchiss v4.1.1 has resolved the issue for me too, thank you - it really is appreciated.

**Comment by aitchiss** [comment]
@reinier I'm looking at your repo and seeing version of 4.0.7 of the plugin being the resolved version in your `yarn.lock` - I can see v4.1.1 defined in your `package.json` though so I'm wondering if the lockfile didn't generate as expected when you upgraded? Can you retry the upgrade of the plugin and make sure the resolved version in the lockfile updates?

**Comment by reinier** [comment]
@aitchiss the deploy with a clear cache worked great! But your hunch about it not working without cache is true unfortunately. Same "Error: Could not find local browser" when deploying with cache after I did the deploy with a clear cache.

**Comment by aitchiss** [comment]
Thank you for confirming! Re-opening this issue so we can investigate the caching problem further - thanks for your patience!

**Comment by arhen** [comment]
Yep, I want to report the same thing. This plugin doesnt work if deploying without clearing cache

**Comment by christopher-theagen** [comment]
@aitchiss Hi Suzanne- might https://github.com/netlify/netlify-plugin-lighthouse/issues/598 be related to this issue?
